### PR TITLE
Fix the "totals" returned by the pagination logic in our cornice services.

### LINKED ISSUE #163: 'total' attribute does not seem to count properly
When I ask for 2 builds of the same update, I always receive `total: 2`, even though only 1 update object is returned. If I set `rows_per_page: 1`, it indicates `pages: 2`, and both pages return the same update object.

```
In [28]:
result = bodhi.query(builds='hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21')
result = munch.unmunchify(result)
result['updates'] = [u['title'] for u in result['updates']]
print result

{u'rows_per_page': 20, u'chrome': True, u'display_user': True, u'pages': 1, u'updates': [u'hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21'], u'total': 2, u'page': 1}

In [30]:
result = bodhi.query(builds='hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21', rows_per_page=1)
result = munch.unmunchify(result)
result['updates'] = [u['title'] for u in result['updates']]
print result

{u'rows_per_page': 1, u'chrome': True, u'display_user': True, u'pages': 2, u'updates': [u'hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21'], u'total': 2, u'page': 1}

In [31]:
result = bodhi.query(builds='hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21', rows_per_page=1, page=2)
result = munch.unmunchify(result)
result['updates'] = [u['title'] for u in result['updates']]
print result

{u'rows_per_page': 1, u'chrome': True, u'display_user': True, u'pages': 2, u'updates': [u'hedgewars-0.9.21.1-3.fc21,ghc-bytestring-show-0.3.5.6-2.fc21'], u'total': 2, u'page': 2}
```

python-fedora-0.5.2-1.fc22.noarch

### PR DESCRIPTION
**Pull request by ralphbean**
This should fix #163.

### CODE REVIEW

File: bodhi/services/user.py [hunk 1]
```diff
@@ -115,7 +116,11 @@ def query_users(request):
         query = query.join(User.packages)
         query = query.filter(or_(*[Package.id==p.id for p in packages]))
 
-    total = query.count()
+    # We can't use ``query.count()`` here because it is naive with respect to
+    # all the joins that we're doing above.
+    count_query = query.statement\
+        .with_only_columns([func.count(distinct(User.id))])\
+        .order_by(None)
```
Comment (lmacken) [inline bodhi/services/user.py]: Missing `total = db.execute(count_query).scalar()` here?

Comment (ralphbean) [inline bodhi/services/user.py]: Got it.  Thanks!

**Comment by lmacken** [comment]
Wow. :boom: :dizzy_face: :zap:  :+1:

**Comment by ralphbean** [comment]
Boom.  THANKS!

**Pull request merged**